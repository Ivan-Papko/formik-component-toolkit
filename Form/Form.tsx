import { Formik } from 'formik'

import { FormikMultiSelect } from './FormikMultiSelect/FormikMultiSelect'
import { FormikAutocomplete } from './FormikAutocomplete'
import { FormikCheckbox } from './FormikCheckbox'
import { FormikCopyInput } from './FormikCopyInput'
import { FormikCountryField } from './FormikCountryField'
import { FormikDatePicker } from './FormikDatePicker'
import { FormikForm } from './FormikForm'
import { FormikPasswordField } from './FormikPasswordField'
import { FormikRegionField } from './FormikRegionField'
import { FormikSelect } from './FormikSelect'
import { FormikSubmit } from './FormikSubmit'
import { FormikSwitch } from './FormikSwitch'
import {
  type IFormikTextFieldProps,
  FormikTextField,
  FormikTextFieldWithRef,
} from './FormikTextField'
import { FormikTimePicker } from './FormikTimePicker'
import { FormikValidatingInput } from './FormikValidatingInput'
import { FormikWrapper } from './FormikWrapper'

/**
 * @deprecated Use `HookForm` instead.
 */
const Form = {
  Context: Formik,
  Form: FormikForm,
  Wrapper: FormikWrapper,
  TextField: FormikTextField,
  TextFieldWithRef: FormikTextFieldWithRef,
  PasswordField: FormikPasswordField,
  Checkbox: FormikCheckbox,
  ValidatingInput: FormikValidatingInput,
  CountryField: FormikCountryField,
  /**
   * @deprecated
   */
  RegionField: FormikRegionField,
  Submit: FormikSubmit,
  Switch: FormikSwitch,
  MultiSelect: FormikMultiSelect,
  Select: FormikSelect,
  Autocomplete: FormikAutocomplete,
  CopyInput: FormikCopyInput,
  TimePicker: FormikTimePicker,
  DatePicker: FormikDatePicker,
}

export { Form, type IFormikTextFieldProps }
export default Form
