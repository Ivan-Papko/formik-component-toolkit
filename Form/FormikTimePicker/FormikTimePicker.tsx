import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import { type TimePickerProps, TimePicker } from '@mui/x-date-pickers/TimePicker'
import { useField } from 'formik'
export interface IFormikTimePickerProps {
  name: string
}

export const FormikTimePicker = ({
  name,
  ...props
}: TimePickerProps<Date> & IFormikTimePickerProps) => {
  const [field, { error }, { setValue }] = useField(name)

  return (
    <FormControl fullWidth error={Boolean(error)}>
      <TimePicker
        value={field.value}
        onChange={value => {
          setValue(value)
        }}
        {...props}
      />
      {error && <FormHelperText>{error}</FormHelperText>}
    </FormControl>
  )
}
