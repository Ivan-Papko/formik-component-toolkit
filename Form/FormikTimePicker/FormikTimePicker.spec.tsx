import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
import dayjs from 'dayjs'
import { object } from 'yup'

import { render } from '@cloudbees/shared-test-utils/render'

import { Form } from '../Form'

import { FormikTimePicker } from './FormikTimePicker'
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
    }
  },
}))

const factory = () =>
  render(
    <LocalizationProvider dateAdapter={AdapterDayjs}>
      <Form.Context
        initialValues={{ time: dayjs('2024-03-18T12:00:00') }}
        onSubmit={jest.fn()}
        validationSchema={object().shape({})}
      >
        <Form.Form>
          <Form.Wrapper>
            <FormikTimePicker
              views={['hours', 'minutes']}
              sx={{ width: '100%' }}
              name="time"
              ampm={false}
              format="HH:mm:ss"
            />
            ,
          </Form.Wrapper>
        </Form.Form>
      </Form.Context>
    </LocalizationProvider>,
    [],
  )

describe('FormikTimePicker', () => {
  it('should match snapshot', () => {
    const { container } = factory()
    expect(container).toMatchSnapshot()
  })

  it('should update date value on change', async () => {
    const { setup, screen } = factory()
    const user = setup()

    const input = screen.getByRole('textbox')
    await user.type(input, '13:30:05', { method: 'userEvent' }) // Typing new date

    expect(input).toHaveValue('13:30:05')
  })
})
