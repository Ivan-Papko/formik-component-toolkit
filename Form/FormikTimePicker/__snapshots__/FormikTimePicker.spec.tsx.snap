// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikTimePicker should match snapshot 1`] = `
.emotion-0 {
  margin-top: 24px;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  width: 100%;
}

.emotion-1 {
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-2 {
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-2 .clearButton {
  opacity: 1;
}

@media (pointer: fine) {
  .emotion-2 .clearButton {
    opacity: 0;
  }

  .emotion-2:hover .clearButton,
  .emotion-2:focus-within .clearButton {
    opacity: 1;
  }
}

.emotion-3 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  padding-right: 14px;
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-3:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-3:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-3.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-3.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-3.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-4 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 16.5px 14px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-right: 0;
}

.emotion-4::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-4:focus {
  outline: 0;
}

.emotion-4:invalid {
  box-shadow: none;
}

.emotion-4::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-4:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-4.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-4:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-4:-webkit-autofill {
  border-radius: inherit;
}

.emotion-5 {
  display: flex;
  height: 0.01em;
  max-height: 2em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  white-space: nowrap;
  color: rgba(0, 0, 0, 0.54);
  margin-left: 8px;
}

.emotion-6 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 50%;
  padding: 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0, 0, 0, 0.54);
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  overflow: visible;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  margin-right: -12px;
}

.emotion-6::-moz-focus-inner {
  border-style: none;
}

.emotion-6.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-6 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-6:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-6:hover {
    background-color: transparent;
  }
}

.emotion-6.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-8 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-9 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-10 {
  float: unset;
  width: auto;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

<div>
  <form
    action="#"
  >
    <div
      class="MuiBox-root emotion-0"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth emotion-1"
      >
        <div
          class="MuiFormControl-root MuiTextField-root emotion-2"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd emotion-3"
          >
            <input
              aria-invalid="false"
              autocomplete="off"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd emotion-4"
              id=":r0:"
              inputmode="text"
              placeholder="hh:mm:ss"
              type="text"
              value="12:00:00"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium emotion-5"
            >
              <button
                aria-label="Choose time, selected time is 12:00 PM"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium emotion-6"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-7"
                  data-testid="ClockIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                  <path
                    d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root emotion-8"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline emotion-9"
            >
              <legend
                class="emotion-10"
              >
                <span
                  class="notranslate"
                >
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      ,
    </div>
  </form>
</div>
`;
